Sources/Core/Video/AVAsset.swift

```diff
@@ -0,0 +1,75 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2015-2021 Alexander Grebenyuk (github.com/kean).
+
+import AVKit
+import Foundation
+
+#if !os(watchOS)
+
+extension ImageContainer {
+    /// A fetched video from data.
+    public var asset: AVAsset? {
+        guard type == .mp4, let data = data else {
+            return nil
+        }
+        return AVDataAsset(data: data)
+    }
+}
+
+// This allows LazyImage to play video from memory.
+final class AVDataAsset: AVURLAsset {
+    private let _loader: DataAssetResourceLoader
```

Comment:
I generally don't use underscores. I think SwiftLint won't like that when I run the CI.